c11threads.h

```diff
@@ -158,10 +162,27 @@ static inline int mtx_trylock(mtx_t *mtx)
 static inline int mtx_timedlock(mtx_t *mtx, const struct timespec *ts)
 {
 	int res;
+#ifdef __APPLE__
+	struct timeval now;
+	struct timespec sleeptime;
+
+	sleeptime.tv_sec = 0;
+	sleeptime.tv_nsec = 5000000; // 5 ms
+
+	while ((res = pthread_mutex_trylock(mtx)) == EBUSY) {
+		gettimeofday(&now, NULL);
 
+		if (now.tv_sec >= ts->tv_sec && (now.tv_usec * 1000) >= ts->tv_nsec) {
```

Comment:
This condition is incorrect. If ts is say 5 sec and 30000 nsec, and now is 15sec and 29999 nsec it will not timeout.

You either have to bring everything into a single common unit and compare once, or do a more complicated condition such as `if(now.tv_sec > ts->tv_sec || (now.tv_sec == ts->tv_sec && (now.tv_usec * 1000) >= ts->tv_nsec)))` ...